cppcache/include/geode/CacheableKey.hpp

```diff
@@ -39,12 +39,10 @@ namespace client {
 /** Represents a cacheable key */
 class APACHE_GEODE_EXPORT CacheableKey : public virtual Cacheable {
  protected:
-  CacheableKey() = default;
   ~CacheableKey() override = default;
```

Comment:
How is it that the destructor being `protected` is not an issue with `std::make_shared`?